src/main/java/trackerbot/task/Task.java

```diff
@@ -0,0 +1,155 @@
+     * </ul>
+     *
+     * @return The '|' delimited String to store in save file.
+     */
+    public abstract String toSaveString();
+
+    /**
+     * Factory method to generate Tasks from the save file.
+     * <p>ofSaveString expects the save string to be split before passing into Task.</p>
+     *
+     * @param type The String representation of the Task in the save file.
+     * @param args The Arguments for each task in the save file.
+     * @return A Task corresponding to the type and args of the save string.
+     * @throws IllegalArgumentException if the save file is corrupted.
+     * @throws DateTimeParseException if the DateTime field is invalid.
+     */
+    public static Task ofSaveString(String type, String... args)
+            throws IllegalArgumentException, DateTimeParseException {
+        switch (type) {
+        case "T":
+            if (args.length != 2) {
+                throw new IllegalArgumentException("Todos should have exactly 2 arguments.");
+            }
+            return new Todo(args);
+        case "D":
+            if (args.length != 3) {
+                throw new IllegalArgumentException("Deadline should have exactly 3 arguments.");
+            }
+            return new Deadline(args);
+        case "E":
+            if (args.length != 4) {
+                throw new IllegalArgumentException("Events should have exactly 4 arguments.");
+            }
+            return new Event(args);
+        default:
+            throw new IllegalArgumentException("Unknown Task type.");
+        }
+    }
+
+    /**
+     * Helper method for toSaveString. <br>
+     * Gets a formatted description and mark status of the Task, and
+     * passes it to the child classes.
+     * @return {Mark Status} | {Description} String, to append to toSaveString
+     *         implementation in child classes.
+     */
+    protected String getSaveInfo() {
+        String checkStatus = isDone ? "1" : "0";
+        return checkStatus + "|" + description;
+    }
+
+    /**
+     * Helper function to determine the checkmark status of the Task.
+     * @return "[X]" if the task is done, "[ ]" otherwise.
+     */
+    private String getCheckbox() {
+        return isDone ? "[X]" : "[ ]";
+    }
+
+    /**
+     * Flags the task to be completed, if able.
+     * @return true if the task is successfully marked to completion,
+     *         false if the task is already completed.
+     */
+    public boolean markTask() {
+        if (isDone) {
+            return false;
+        }
+        isDone = true;
+        return true;
+    }
+
+    /**
+     * Flags the task to be incomplete, if able.
+     * @return true if the task is successfully marked to be incomplete,
+     *         false if the task is already marked as incomplete.
+     */
+    public boolean unmarkTask() {
+        if (!isDone) {
+            return false;
+        }
+        isDone = false;
+        return true;
+    }
```

Comment:
Given that this is returning a boolean, it would be good to name it to sound like a boolean in accordance to the Java coding standard by our course. 

```suggestion
    public boolean canMarkTask() {
        if (isDone) {
            return false;
        }
        isDone = true;
        return true;
    }

    /**
     * Flags the task to be incomplete, if able.
     * @return true if the task is successfully marked to be incomplete,
     *         false if the task is already marked as incomplete.
     */
    public boolean canUnmarkTask() {
        if (!isDone) {
            return false;
        }
        isDone = false;
        return true;
    }
```

Reply:
Hmm, should this really be named this way? These methods actually mark/unmark the task directly, and the boolean flag is actually to feedback to the caller whether the task has been marked/unmarked successfully.